docker/generate-config.js

```diff
@@ -1,68 +1,73 @@
 const fs = require("fs");
 const path = require("path");
 
-const configPath = path.resolve("/usr/share/nginx/html/config.js");
+const htmlPath =
+  process.env.NODE_ENV === "development"
+    ? path.resolve(__dirname, "..", "public")
+    : "/usr/share/nginx/html";
+const configPath = path.resolve(htmlPath, "config.js");
+const themesPath = path.resolve(htmlPath, "themes");
 
 console.log("Pre-start: Generating config file…");
 
-let content = "";
-try {
-  content = fs.readFileSync(configPath, { encoding: "utf-8" });
-} catch {
-  console.log(
-    `Pre-start: ⚠️ No existing config file found under "${configPath}".`
-  );
-}
-
-const convertToNativeType = (input) => {
-  if (["true", "false"].includes(input)) {
-    return input === "true";
-  }
+const config = readConfig();
 
-  if (!Number.isNaN(+input)) {
-    return +input;
+Object.entries(process.env).forEach(([key, value]) => {
+  if (key.startsWith("VOCASCAN_")) {
+    config[key.replace(/^VOCASCAN_/, "")] = convertToNativeType(value);
   }
+});
 
-  return input;
-};
+if (fs.existsSync(themesPath) && fs.lstatSync(themesPath).isDirectory()) {
+  const themes = Object.fromEntries(
+    fs
+      .readdirSync(themesPath)
+      .map((theme) => [
+        theme.replace(/(themes\/|.css)/g, ""),
+        `themes/${theme}`,
+      ])
+  );
+  config.themes = { ...config.themes, ...themes };
+}
 
-const envVars = Object.entries(process.env).reduce((acc, [key, value]) => {
-  if (key.startsWith("VOCASCAN_")) {
-    acc[key.replace(/^VOCASCAN_/, "")] = convertToNativeType(value);
-  }
+writeConfig(config);
 
-  return acc;
-}, {});
+console.log(
+  `Pre-start: ✓ Successfully written config file to "${configPath}".`
+);
 
-const existing = content.match(/window\.VOCASCAN_CONFIG *= *(\{[\s\S]*\})/);
-let existingConfig = {};
+function readConfig() {
```

Comment:
Can you use arrow functions? (this includes all functions you defined)

Reply:
Arrow functions are not hoisted which means that they have to be written at the top of the file. I want to put the functions below the code for readability reasons.

Reply:
Done